workers/unity/Packages/com.improbable.gdk.core/Tests/Editmode/GameObjectRepresentation/SpatialOSComponentTests.cs

```diff
@@ -81,5 +82,47 @@ public void TryGetGameObjectForEntity_returns_false_when_entity_does_not_exist()
             Assert.IsFalse(succeeded);
             Assert.IsNull(linkedGameObject);
         }
+
+        [Test]
+        public void TryGetSpatialOSEntityIdForGameObject_returns_true_when_gameobject_is_linked()
+        {
+            var component = testGameObject.GetComponent<SpatialOSComponent>();
+            Assert.NotNull(component);
+
+            var succeeded = component.TryGetSpatialOSEntityIdForGameObject(testGameObject, out var linkedSpatialEntityId);
+            Assert.IsTrue(succeeded);
+            Assert.AreEqual(testSpatialEntityId, linkedSpatialEntityId);
+        }
+
+        [Test]
+        public void TryGetSpatialOSEntityIdForGameObject_returns_false_when_gameobject_is_not_linked()
+        {
+            worker.EntityIdToEntity.Add(testSpatialEntityId, testEntity);
+            var component = testGameObject.GetComponent<SpatialOSComponent>();
+            Assert.NotNull(component);
+
+            var succeeded = component.TryGetSpatialOSEntityIdForGameObject(new GameObject(), out var linkedSpatialEntityId);
+            Assert.IsFalse(succeeded);
+            Assert.AreEqual(default(EntityId), linkedSpatialEntityId);
+        }
+
+        [Test]
+        public void TryGetSpatialOSEntityIdForGameObject_returns_false_when_entity_on_different_worker()
+        {
+            worker.EntityIdToEntity.Add(testSpatialEntityId, testEntity);
+            var component = testGameObject.GetComponent<SpatialOSComponent>();
+            Assert.NotNull(component);
+
+            var testGameObject2 = new GameObject();
```

Comment:
nitpick - would prefer a name that makes it clear what this object is. E.g. `gameObjectOnDifferentWorker`.